source/base/discrete_time.cc

```diff
@@ -66,6 +66,16 @@ DiscreteTime::set_desired_next_step_size(const double next_step_size)
 
 
 
+void
+DiscreteTime::set_next_step_size(const double next_step_size)
+{
+  Assert(next_step_size > 0, ExcInternalError());
+  next_time = current_time + next_step_size;
+  Assert(next_time <= end_time, ExcInternalError());
```

Comment:
I didn't spot this at first - my mistake. This isn't an internal error - its an error created by incorrect user input. Would you please switch these exceptions to use `ExcMessage`?